### PR TITLE
 #13681 - Allow null $keyPath on Hash::combine and use numbered array if it is

### PR DESCRIPTION
**Pull request by victoreassis**
This should solve issue #13681 

It's my first contribution though, so i don't know if it's a perfect one.

### CODE REVIEW

File: src/Utility/Hash.php [hunk 1]
```diff
@@ -466,10 +466,12 @@ public static function combine(array $data, $keyPath, $valuePath = null, $groupP
         if (is_array($keyPath)) {
             $format = array_shift($keyPath);
             $keys = static::format($data, $keyPath, $format);
+        } elseif (is_null($keyPath)) {
```
Comment (dereuromark) [inline src/Utility/Hash.php]: `$keyPath === null`
and others

**Comment by dereuromark** [comment]
You might want to switch the target to 3.next here (master is 3.current = 3.8 and for bugfixes only).

File: src/Utility/Hash.php (hunk 1 above)
Comment (victoreassis) [reply src/Utility/Hash.php]: Ok. I will do that.

**Comment by victoreassis** [comment]
Got it.

**Comment by dereuromark** [comment]
I managed to do it for you

**Comment by victoreassis** [comment]
Thanks

File: src/Utility/Hash.php (hunk 1 above)
Comment (victoreassis) [reply src/Utility/Hash.php]: Done.

**Comment by dereuromark** [comment]
This change requires some test cases in HashTest class.

**Comment by dereuromark** [comment]
Please note: Your Git credentials is not set up correctly locally. You are commiting and pushing here as anonymous user.

**Comment by victoreassis** [comment]
Ok. I will check the HashTest and the credentials.

**Comment by victoreassis** [comment]
I can't tell what exactly happened on the check that failed. It appears to have passed tests but failed anyway.

**Comment by dereuromark** [comment]
Great work!
The appv build can be ignored for now.

**Comment by victoreassis** [comment]
Thanks.

I have a question, the method description shouldn't be updated to include array as a possible value of $keyPath and $valuePath? As they can receive one to format them.

**Comment by markstory** [comment]
Thank you 👍

**Comment by andrii-pukhalevych** [comment]
Does it preserve keys from source array or just generates new numeric indexes?

**Comment by victoreassis** [comment]
It generates new ones.

**Pull request merged**